bioscrape/inference_setup.py

```diff
@@ -506,49 +512,33 @@ def plot_mcmc_results(self, sampler, plot_show = True, **kwargs):
             axes[-1].set_xlabel("step number")
         else:
             axes.set_xlabel("step number")
+        figure_objects.append(fig)
+        figure_objects.append(axes)
 
-        if 'discard' in kwargs.keys():
-            discard = kwargs.get('discard')
-        else:
-            discard = 100 #arbitrarily discard the first 100 steps
-        if 'thin' in kwargs.keys():
-            thin = kwargs.get('thin')
-        else:
-            if convergence_check:
```

Comment:
Have you removed all convergence checks? I personally found this feature annoying because it would just fail without returning anything so often a lot of computation would be wasted if convergence check were true.

Reply:
I didn't remove them but changed them quite a bit so that they are not hard-coded. Also, I set `convergence_check = False` to be the default since I also felt that it is annoying to lose the whole run just because emcee thinks it did not converge. User can now set `convergence_check = True` if they need to compute autocorrelation times in their MCMC runs.